cmd/kubeadm/app/cmd/join.go

```diff
@@ -62,6 +63,10 @@ func NewCmdJoin(out io.Writer, s *kubeadmapi.KubeadmConfig) *cobra.Command {
 
 // RunJoin executes worked node provisioning and tries to join an existing cluster.
 func RunJoin(out io.Writer, cmd *cobra.Command, args []string, s *kubeadmapi.KubeadmConfig) error {
+	// Require root and that kubelet and docker is running
+	if err := kubeadmpreflight.Verify(); err != nil {
```

Comment:
I'd like this to be a separate command also. May be for the moment we should only provide this functionality as a separate command.